### PR TITLE
Add logs all over the place

### PR DESCRIPTION
**Pull request by ceyhunsen**
# Description

This PR adds log statements, all over the place. For debugging purposes.

For the places `#[tracing::instrument(skip(self), err(level = tracing::Level::ERROR), ret(level = tracing::Level::TRACE))]` is used for a function, it should be excepted to see a log containing arguments and return value of that function after the function is returned:

```bash
2024-09-10T13:42:53.248374Z TRACE sign_taproot_script_spend_tx_new_tweaked{tx_handler=TxHandler { tx: Transaction { version: Version(2), lock_time: 0 blocks, input: [TxIn { previous_output: OutPoint { txid: 49d799e7cb2fa5c20ed92340864ab189e683e04f95bf3d6290f471f9b31960f3, vout: 0 }, script_sig: Script(), sequence: Sequence(0xfffffffd), witness: Witness: { indices: 0, indices_start: 0, witnesses: [] } }], output: [TxOut { value: 330 SAT, script_pubkey: Script(OP_PUSHNUM_1 OP_PUSHBYTES_32 70fe0a680d3ab666f78aa0bfaf2ad9f45d4682ddb96cf5e57efff281e935b7b5) }] }, prevouts: [TxOut { value: 1000 SAT, script_pubkey: Script(OP_PUSHNUM_1 OP_PUSHBYTES_32 70fe0a680d3ab666f78aa0bfaf2ad9f45d4682ddb96cf5e57efff281e935b7b5) }], scripts: [[Script(OP_PUSHBYTES_32 f8e8579c126f49ded337c19c4f5f1c1951f0752162d6a61f0a9e15585594394b OP_CHECKSIG)]], taproot_spend_infos: [TaprootSpendInfo { internal_key: XOnlyPublicKey(51def5d05ddd28061f3dd004f860cee7baf4c8f7c4218844758a51968d37c793280f903bc47f922b1c4d4fdcb73801679003a5ac7112f6374c8b41f58d48e6b7), merkle_root: Some(753b565398b9da7f6e325091c453fddcd4bf79f48b481f4557d8a211ce823e02), output_key_parity: Even, output_key: TweakedPublicKey(XOnlyPublicKey(b5b735e981f2ff7ee5f56cb9dd82465df4d92aafbfa08af766b63a0d680afe70bcba5b825134439cee018f146dbb59589ba1c6ecff966284167a9283ae612bae)), script_map: {(Script(OP_PUSHBYTES_32 f8e8579c126f49ded337c19c4f5f1c1951f0752162d6a61f0a9e15585594394b OP_CHECKSIG), TapScript): {TaprootMerkleBranch([])}} }] } txin_index=0 script_index=0}:sign_with_tweak{sighash=168d96a499e0ae147f4a50b09ab184c94288125b368d29462060a841f5bff900 merkle_root=None}: clementine_core::actor: return=Signature(52906a0af00a2ff9b6cc83ea4ca9f99366f407d8658025f1647c53451a36c65d5def44fcdc2227134ddbdddc1871c8215acdc29ffeb0d73a28419ef324175c8f)
```

### CODE REVIEW

**Comment by ceyhunsen** [comment]
Unnecessary function logs should be removed. If there is, please report.

File: core/src/actor.rs [hunk 1]
```diff
@@ -20,10 +20,17 @@ pub struct Actor {
 }
 
 impl Actor {
+    #[tracing::instrument(ret(level = tracing::Level::TRACE))]
     pub fn new(sk: SecretKey, network: bitcoin::Network) -> Self {
         let keypair = Keypair::from_secret_key(&utils::SECP, &sk);
         let (xonly, _parity) = XOnlyPublicKey::from_keypair(&keypair);
         let address = Address::p2tr(&utils::SECP, xonly, None, network);
+        tracing::trace!(
+            "Creating a new actor with keypair {:?}, x-only public key {:?} and address {:?}",
```
Comment (ozankaymak) [inline core/src/actor.rs:29]: Printing keypair is not necessary

File: core/src/actor.rs [hunk 2]
```diff
@@ -76,9 +87,12 @@ impl Actor {
             TapLeafHash::from_script(spend_script, LeafVersion::TapScript),
             bitcoin::sighash::TapSighashType::Default,
         )?;
+        tracing::trace!("Signature hash is {:?}", sig_hash);
```
Comment (ozankaymak) [inline core/src/actor.rs:90]: Unnecessary IMO

**Comment by ceyhunsen** [comment]
Looking at printlns. They should be replaced with tracing.

**Comment by ceyhunsen** [comment]
Couldn't found any printlns in any place other than tests and binaries. Can you give pointers?

**Comment by ozankaymak** [comment]
If printlns are OK in the binaries, then it is fine.

**Review by ozankaymak — APPROVED** [review]

**Review by ekrembal — APPROVED** [review]

**Pull request merged**